### PR TITLE
[meta] GetClass better handling of typedef autoloading.

### LINKED ISSUE #12378: GetClass(<typedef>) works only at the second try
### Describe the bug
We have typedefs in ATLAS EDM that are not resolved immediately by TClass::GetClass() but only at the second call to it:
root [0] TClass::GetClass("MissingETBase::Types::jetlink_t")
(TClass *) nullptr
root [1] TClass::GetClass("MissingETBase::Types::jetlink_t")
(TClass *) 0x4d3eff0

I am tentatively calling it a bug - because the behavior is inconsistent, but it's also possible we are missing something in selection.xml files?

### Expected behavior
Hopefully it would work at the first try!  😄 

### To Reproduce
For ROOT 6.26.8  ATLAS setup, on lxplus:
% source /cvmfs/atlas.cern.ch/repo/ATLASLocalRootBase/x86_64/AtlasSetup/V02-01-05/AtlasSetup/scripts/asetup.sh master,latest,Athena
% root

For ROOT master  ATLAS setup (NOTE-the build is ongoing, wait till later or tomorrow):
source /cvmfs/atlas.cern.ch/repo/ATLASLocalRootBase/x86_64/AtlasSetup/V02-01-05/AtlasSetup/scripts/asetup.sh master--dev3LCG,latest,Athena

### Additional context
The typedef is defined in a different library than the actual class - I suspect the first call to GetClass() only loads the first library, gets the typedef definition but stops at this point. Only the second call loads the other library with the real class definition.

This is not a stopper for us, but kind of annoying.

"Typedef package" includes and selection.xml:
https://acode-browser1.usatlas.bnl.gov/lxr/source/athena/Event/xAOD/xAODMissingET/xAODMissingET/
"Class package" includes:
https://acode-browser1.usatlas.bnl.gov/lxr/source/athena/Event/xAOD/xAODJet/xAODJet/

Cheers, Marcin

### PR DESCRIPTION
**Pull request by pcanal**
Handle the case where a typedef as a dictionary entry and thus an autoload entry (rootmap case) and this typedef point to an instance a class template with a default paramter.  In this case we do not have a (forward) declaration of the typedef in the rootmap file (because we can't have a forward declaration of the class template).  In those case our only source of information is in the list of type which is loaded as part of the dictionary for the typedef.  If in addition the dictionary for the target of the typedef is in another library we actually need to load that library based on the resolved name of the typedef).

This fixes #12378

### CODE REVIEW

File: core/meta/src/TClass.cxx [hunk 1]
```diff
@@ -3102,6 +3102,17 @@ TClass *TClass::GetClass(const char *name, Bool_t load, Bool_t silent, size_t hi
...
+         // information might be pertinent ot the normalization of the name.
+         // For example it might contain or be a typedef for which we don't
+         // have a forward declaration (eg. typedef to instance of class
+         // template with default paramters).  So let's redo the normalization
+         // as the new information (eg typedef in TROOT::GetListOfTypes) might
+         // lead to a different value.
+         {
+            TInterpreter::SuspendAutoLoadingRAII autoloadOff(gInterpreter);
+            TClassEdit::GetNormalizedName(normalizedName, name);
+         }
```
Comment (jalopezg-git) [inline core/meta/src/TClass.cxx:3115]: IIUC, this tries to normalize the type name again (first try in https://github.com/root-project/root/blob/47e277a3fb88e8845eebf90bca207f0c3bed904d/core/meta/src/TClass.cxx#L3053-L3056), in case new information became available.

However, what if we have a `typedef` that aliases a `typedef` (that aliases ...), that names a class.  Would that require additional steps?

Comment (jalopezg-git) [reply core/meta/src/TClass.cxx:3115]: Also, unrelated to the additions in the patch: `TClass:GetClass()` code seems already hard to follow (incl. special handling for some cases).  This adds to it.
I think it should be refactored at some point (?)

Comment (pcanal) [reply core/meta/src/TClass.cxx:3115]: > IIUC, this tries to normalize the type name again

Yes, that the main point of this PR.

> However, what if we have a typedef that aliases a typedef (that aliases ...), that names a class. Would that require additional steps?

It might but not.  Presumably all the dependent library with loaded during the "explicit" autoload right above.  And no new libraries would be loaded as part of this line because of the suspension.
